src/app/search/query-parser.ts

```diff
@@ -382,6 +398,21 @@ export function* lexer(query: string): Generator<Token> {
   }
 }
 
+/**
+ * Quote a string if it's needed.
+ *
+ * @example
+ *
+ * quoteFilterString("foo bar") => "\"foo bar\""
+ * quoteFilterString("foobar") => "foobar"
+ * quoteFilterString("foo\"bar") => foobar"
+ */
+export function quoteFilterString(arg: string) {
+  const quoteChar = arg.includes('"') ? "'" : '"';
+  arg = arg.replaceAll(quoteChar, `\\${quoteChar}`);
```

Comment:
Any loadout names with a backslash in their names won't be escaped correctly. Would be nice to have a round-trip test that quotes some esoteric loadout names, parses the result, and checks that the name matches.